native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2ConnectionManager.java

```diff
@@ -147,6 +158,35 @@ void removeClientChannel(HttpRoute httpRoute, Http2ClientChannel http2ClientChan
         }
     }
 
+    void markClientChannelAsStale(HttpRoute httpRoute, Http2ClientChannel http2ClientChannel) {
+        Http2ChannelPool.PerRouteConnectionPool perRouteConnectionPool = fetchPerRoutePool(httpRoute);
+        if (perRouteConnectionPool != null) {
+            perRouteConnectionPool.removeChannel(http2ClientChannel);
+        }
+        http2ClientChannel.setTimeSinceMarkedAsStale(System.currentTimeMillis());
+        http2StaleClientChannels.add(http2ClientChannel);
+    }
+
+    private void initiateConnectionEvictionTask() {
```

Comment:
Isn't there any external library which provide this eviction functionality? Any particular reason to implement this on our own?

Reply:
we evict targetchannel based on the same configs and same functionality, but this is handled buy the apache GenericObjectPool. We cannot use that in our case.